### DISCUSSION TITLE
Charger switches to "Plugged in" despite being actually Unplugged

### DISCUSSION THREAD

**Discussion by thuesen** [alandtse/tesla#781] [Q&A] [answered]
Edit: version 3.19.3.

I noticed that the charger binary_sensor switched to "Plugged in" while car was not plugged in.

I used the "Force data update" button and after a few seconds, charger would correctly switch to "Unplugged".  After another few seconds, it switched wrongly to "Plugged in" again.

I let it be and it went like this several times, every 30 minutes (my polling interval). 

![Screenshot 2023-11-16 at 15 13 43](https://github.com/alandtse/tesla/assets/4000048/866c841f-bf36-413b-93d9-bdff3fe24253)

On the screenshot I unplugged charger around 8:30 and HA picked it up around 8:45 that charger was unplugged. And then around 12:30 it started plugging in. Then unplugged for a few seconds, plugged in again. This went on until 14:30.  

Car has been parked, and unplugged, for the entire time.

Does the integration default to "Plugged in" or what could be the reason for this?

**Comment by skipishere** [comment, marked as answer]
Would be useful to know what version of the integration you currently are using, in https://github.com/alandtse/tesla/issues/758 it's mentioned that this is taken from the Tesla API reported state. Does your Tesla app also think it is plugged in?

**Reply by thuesen** [reply]
Sorry for leaving that out.  I'm using 3.19.3.  I had searched beforehand but failed to find issue #758 which does sound like my issue as well.


I loaded up my Tesla app the first time I saw this issue today, and the app correctly said it was NOT plugged in, which is why I found it odd that the integration said "plugged in".  Force updating the integration also correctly changed to "Unplugged", as mentioned, but it changes back again after a few seconds to the wrong state.

My main issue is that when the charger is plugged in, this also tells EV Smart Charging that the car is ready to to be scheduled for charging when the power is cheap, as by this recommendation: https://github.com/jonasbkarlsson/ev_smart_charging#example-of-automation-to-inform-the-integration-that-the-ev-is-connected-to-the-charger

**Reply by skipishere** [reply]
Might be worth raising/adding to that issue with logs showing the the state change occurring.

**Comment by MAFL74** [comment]
@thuesen Did you ever manage to solve this issue?  I have the exact same problem in my integration. Keeps flapping Plugged in/Unplugged.

**Reply by thuesen** [reply]
I'm afraid I don't remember what I did to solve it - if anything. But I don't seem to have the issue anymore.
